e2e/test/helpers/TestDevice.cs

```diff
@@ -112,6 +112,18 @@ await RetryOperationHelper
                     s_logger)
                 .ConfigureAwait(false);
 
+            // Confirm the device exists in the registry before calling it good to avoid downstream test failures.
+            await RetryOperationHelper
```

Comment:
Prevent cases where we create a device but then get it and we get null back from the service.